### ISSUE TITLE
🐛 BUG: Assign environment variables to .ts file during build time

### ISSUE THREAD

**Issue by huydoansg** [snowpackjs/astro#1397]
### What package manager are you using?

npm

### What operating system are you using?

Mac

### Describe the Bug

I configured some variables in the .env file. I can get it in .astro file when build time by console.log.

But cannot assign these variables in .env file into .ts file as constants variable to use later in runtime

Do you know why?
I guess snowpack or astro does not assign during build time for .ts file

### Steps to Reproduce

1. install astro with vue renderer
2. install dotenv plugin and configure in `snowpack.config.mjs` file
// snowpack.config.mjs
// Example: alias types
export default {
  alias: {
    '@components': './src/components',
    '@core': './src/core',
  },
  plugins: [
    '@snowpack/plugin-dotenv',
    '@snowpack/plugin-typescript'
  ],
};
3. configure .env file
SNOWPACK_PUBLIC_ENABLE_FEATURE=true
SNOWPACK_PUBLIC_API_ENPOINT=https://my-api.io

**Comment by huydoansg** [comment]
See this ``Remember: that these env variables are statically injected into your application for everyone at build time, and not runtime.`` from document.

It looks like the limitation

**Comment by coelmay** [comment]
Have you tried using [snowpack env](https://www.snowpack.dev/reference/configuration#env) @huydoansg?
E.g.
```javascript
// snowpack.config.mjs
export default {
  env: {
    API_URL: 'api.google.com',
  },
};
```
then access via `import.meta.env` (e.g. `import.meta.env.API_URL`.)

**Comment by huydoansg** [comment]
> Have you tried using [snowpack env](https://www.snowpack.dev/reference/configuration#env) @huydoansg?
> E.g.
> 
> ```js
> // snowpack.config.mjs
> export default {
>   env: {
>     API_URL: 'api.google.com',
>   },
> };
> ```
> 
> then access via `import.meta.env` (e.g. `import.meta.env.API_URL`.)

I think the answer is here:

`See this Remember: that these env variables are statically injected into your application for everyone at build time, and not runtime. from document.`

**Comment by coelmay** [comment]
> But cannot assign these variables in .env file into .ts file as constants variable to use later in runtime

So what is the bug? The issue title is **"BUG: Assign environment variables to .ts file during build time"**, yet you seemingly wish to have them available at runtime. So is the issue really that you can't access these variables at runtime? If so, perhaps you need to change the issue name.

**Comment by FredKSchott** [comment]
Thanks for the report! We believe that this has been fixed in astro v0.21, going out today. Let us know if you can still reproduce the issue after updating to v0.21 and we'll be happy to take a look.
